crates/recursor/src/error.rs

```diff
@@ -195,31 +195,31 @@ impl From<ResolveError> for Error {
             Err(e) => return ErrorKind::Resolve(e).into(),
         };
 
-        let ProtoErrorKind::NoRecordsFound {
-            query,
-            soa,
-            ns,
-            authorities,
-            ..
-        } = proto_err
-        else {
-            return ErrorKind::Message("unexpected error kind: not no records found").into();
-        };
-
-        if let Some(ns) = ns {
-            ErrorKind::ForwardNS(ns).into()
-        } else if let Some(soa) = soa {
-            ErrorKind::Forward(ForwardData::new(
+        match proto_err {
```

Comment:
Could we not yield `ErrorKind::Proto(proto_err.into()).into()` from the `else` in the previous `let-else` construction? The reduction of rightward drift was nice.